src/server/api/geo/getForestAgreementLayer.ts

```diff
@@ -6,33 +6,54 @@ import { ForestSource } from '@meta/geo'
 import { GeoController } from '@server/controller/geo'
 import Requests from '@server/utils/requests'
 
-export const getForestAgreementLayer = async (req: Request, res: Response) => {
+export const getForestAgreementLayer = async (
+  req: Request<
+    { countryIso: CountryIso },
+    never,
+    never,
+    { gteAgreementLevel: number; gteHansenTreeCoverPerc: number; layer: Array<string>; opacity: number }
+  >,
+  res: Response
+) => {
   try {
-    const countryIso: CountryIso = req.params.countryIso as CountryIso
-    const gteAgreementLevel = Number(req.query.gteAgreementLevel)
-    const gteHansenTreeCoverPerc = Number(req.query.gteHansenTreeCoverPerc)
-
+    const { countryIso } = req.params
+    const { gteAgreementLevel, gteHansenTreeCoverPerc, layer, opacity } = req.query
     const sourceOptions = Object.values(ForestSource)
 
-    if (Array.isArray(req.query.layer) && req.query.layer.length > 1) {
+    if (Array.isArray(layer) && layer.length > 1) {
       const sourceLayersSet: Set<ForestSource> = new Set([])
-      req.query.layer.forEach((layer) => {
-        if (sourceOptions.includes(layer as unknown as ForestSource))
-          sourceLayersSet.add(ForestSource[layer as unknown as ForestSource])
-        else throw Error(`Not valid forest source for agreement-${layer}`)
+
+      layer.forEach((layerId: string) => {
+        if (sourceOptions.includes(layerId as unknown as ForestSource)) {
+          if (
+            layerId === ForestSource.Hansen &&
+            (gteHansenTreeCoverPerc === undefined ||
+              Number.isNaN(Number(gteHansenTreeCoverPerc)) ||
+              gteHansenTreeCoverPerc < 0 ||
+              gteHansenTreeCoverPerc > 100)
+          )
+            throw Error(`Not valid Hansen tree cover percentage 0-100: ${gteHansenTreeCoverPerc}`)
```

Comment:
If you `throw Error` I think the whole node/express server shuts down
Better to use `Requests.sendErr`

Reply:
In these cases the errors are catched below and Request.sendErr is called